### DISCUSSION TITLE
Lack of earth layer

### DISCUSSION THREAD

**Discussion by pkaminski2108** [systemed/tilemaker#584] [Q&A] [answered]
I'd like to add an 'earth' layer (similar to what is done here: https://protomaps.github.io/PMTiles/?url=https%3A%2F%2Fr2-public.protomaps.com%2Fprotomaps-sample-datasets%2Fprotomaps-basemap-opensource-20230408.pmtiles#map=0.52/0/0).

However, after adding this item to config.json
`"earth": { "minzoom": 0, "maxzoom": 14 },`

and adding this to process.lua
```
local earth = node:Find("earth") 
if earth ~= "" then 
  node:Layer("earth", false) 
  node:Attribute("class", earth) 
  SetNameAttributes(node) 
  return 
end
```

The 'earth' layer is empty - could someone help me with this?

I am not uploading the entire files because I am based on .`/resources/config-openmaptiles.json` and `/resources/process-openmaptiles.lua.`

**Comment by systemed** [comment]
Your code will look for nodes in the source OSM data with an "earth" tag, and write them out to your vector tiles. That's probably not going to work - nodes tagged as "earth" are not a thing in OSM.

To add a new layer, generally you'll need to identify your source data (which may be something tagged in OSM or external shapefiles), then write the Lua script to recognise this and write out vector data accordingly. I'm not sure what the Protomaps earth layer contains but that should be your starting point.

**Reply by pkaminski2108** [reply]
So @systemed I wonder what I did wrong there (image below)? I thought I was missing the earth layer because in this example [LINK](https://protomaps.github.io/PMTiles/?url=https%3A%2F%2Fr2-public.protomaps.com%2Fprotomaps-sample-datasets%2Fprotomaps-basemap-opensource-20230408.pmtiles#map=2.25/37.96/-55) the earth layer (maybe it's just a name) is present and the countries and their borders are visible

<img width="783" alt="PS" src="https://github.com/systemed/tilemaker/assets/51911742/a4e2d4c0-1d43-4ea1-bc7b-1a4e373f340d">

**Reply by systemed** [reply]
What are you expecting to see in the Earth layer?

**Reply by pkaminski2108** [reply]
If you look at the example to which I provided the link, you can see the country borders + the background of the countries stands out on the map (it has a different color). You can't see it on my screenshot - the background of the countries blends with the oceans. I thought this was what the earth layer is responsible for

**Reply by systemed** [reply, marked as answer]
If you create a map with tilemaker using the default (OpenMapTiles-compatible) schema, and the default style, country boundaries are included.

You don't appear to have the ocean layer in your map - possibly you haven't included the water shapefiles as described in the readme?

**Comment by BenOnTrack** [comment]
Hi @pkaminski2108 , have you found a decent setting for 
`"ocean":             { "minzoom": 0,  "maxzoom": 14, "source": "coastline/water_polygons.shp", "simplify_below": 13, "simplify_level": 0.0001, "simplify_ratio": 2, "write_to": "water", "index": true }`

When I run that current setting (from config-openmaptiles), on a small country (NZ) i end up with a 5gb mbtiles.

**Reply by BenOnTrack** [reply]
aw actually it looks like this config has changed since my last pull - going to try with the new config
`"ocean":             { "minzoom": 0,  "maxzoom": 14, "source": "coastline/water_polygons.shp", "filter_below": 12, "filter_area": 0.5, "simplify_below": 13, "simplify_level": 0.0001, "simplify_ratio": 2, "write_to": "water" },`

Edit: still 5Gb
